src/main/java/picard/nio/PathHelper.java

```diff
@@ -0,0 +1,42 @@
+package picard.nio;
+
+import picard.cmdline.ClassFinder;
+
+/**
+ * Created by farjoun on 11/13/17.
+ */
+public class PathHelper {
+
+    private PathHelper() {}
+
+    public static boolean isHasGoogle() {
+        return hasGoogle;
+    }
+
+    private static boolean hasGoogle = false;
+
+    public static void initilizeAll() {
+        ClassFinder finder = new ClassFinder();
+        finder.find("",Object.class);
+        // google
+
+        if (isGoogleOnPath()) {
+            hasGoogle = true;
+            initGoogle();
+        }
+    }
+
+    private static void initGoogle() {
+        GoogleStorageUtils.initialize();
+    }
+
+    private static boolean isGoogleOnPath() {
+        try {
+
+            new GoogleStorageUtils();
+            return true;
+        } catch (NoClassDefFoundError e) {
+            return false;
+        }
```

Comment:
I am not a big fan of all these named methods for google, since this limits how we would add additional nio implementations in the future.  Could you instead have an enum for each supported nio plugin, and then change the methods to take in one or more enum values to either test for (`onPath(Enum provider...)`), initialize (`init(Enum provider..)`), or if supported (`supported(Enum provider..)`).

Reply:
good idea. thanks.